doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,430 @@
+between them based on the version number provided, see `CMake package version`_ for details.
+
+For example:
+
+.. code-block:: none
+
+   <projects>/zephyr-workspace-2.a
+   └── zephyr                       (Exported to CMake)
+
+   <projects>/zephyr-workspace-2.b
+   └── zephyr                       (Exported to CMake)
+
+   <home>/app
+   ├── CMakeLists.txt
+   ├── prj.conf
+   └── src
+       └── main.c
+
+in this case, there are two released versions of Zephyr installed at their own workspaces.
+Workspace 2.a and 2.b, corresponding to the Zephyr version.
+
+To ensure ``app`` is built with minimum version ``2.a`` the following ``find_package``
+syntax may be used:
+
+.. code-block:: cmake
+
+   cmake_minimum_required(VERSION 3.13.1)
+   find_package(Zephyr 2.a)
+   project(app)
+
+
+Note that both ``2.a`` and ``2.b`` fulfill this requirement.
+
+CMake also supports the keyword ``EXACT``, to ensure an exact version is used, if that is required.
+In this case, the application CMakeLists.txt could be written as:
+
+.. code-block:: cmake
+
+   cmake_minimum_required(VERSION 3.13.1)
+   find_package(Zephyr 2.a EXACT)
+   project(app)
+
+In case no Zephyr is found which satisfies the version required, as example, the application specifies
+
+.. code-block:: cmake
+
+   cmake_minimum_required(VERSION 3.13.1)
+   find_package(Zephyr 2.z)
+   project(app)
+
+then an error similar to below will be printed::
+
+  Could not find a configuration file for package "Zephyr" that is compatible
+  with requested version "2.z".
+
+  The following configuration files were considered but not accepted:
+
+    <projects>/zephyr-workspace-2.a/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake, version: 2.a.0
+    <projects>/zephyr-workspace-2.b/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake, version: 2.b.0
+
+
+.. note:: It can also be beneficial to specify a version number for Zephyr repository applications
+          and Zephyr workspace applications. Specifying a version in those cases ensures the
+	  application will only build if the Zephyr repository or workspace is matching.
+	  This can be useful to avoid accidental builds when only part of a workspace has been
+	  updated.
+
+
+Multiple Zephyr Installations (Zephyr workspace)
+************************************************
+
+Testing out a new Zephyr version, while at the same time keeping the existing Zephyr in the
+workspace untouched is sometimes beneficial.
+
+Or having both an upstream Zephyr, Vendor specific, and a custom Zephyr in same workspace.
+
+For example:
+
+.. code-block:: none
+
+   <projects>/zephyr-workspace
+   ├── zephyr
+   ├── zephyr-vendor
+   ├── zephyr-custom
+   ├── ...
+   └── my_applications
+        └── my_first_app
+
+
+in this setup, ``find_package(Zephyr)`` has the following order of precedence for selecting
+which Zephyr to use:
+
+* Project name: ``zephyr``
+* First project, when Zephyr projects are ordered lexicographical, in this case.
+
+  * ``zephyr-custom``
```

Comment:
Should this list also contain "zephyr"?

Reply:
As the `zephyr` itself has the highest preference per default, then there is little point is adding `zephyr` itself to the list.

I prefer to keep it simple, by telling users how the can make other `zephyr`-folders higher preference.